boost_library_docs_tracker/services.py

```
@@ -0,0 +1,124 @@
+    Get or create a BoostDocContent row for the given URL.
+    Page content is NOT stored in the DB; it lives in workspace files.
+
+    Returns (doc_content, change_type) where change_type is one of:
+      "created"         — URL was not in DB; row inserted.
+      "content_changed" — URL exists but content_hash differs; hash updated.
+      "unchanged"       — URL exists and content_hash matches; only scraped_at updated.
+
+    Raises ValueError if url is empty.
+    """
+    if not url or not url.strip():
+        raise ValueError("url must not be empty")
```

Ruff (0.15.4)

[warning] Avoid specifying long messages outside the exception class

(TRY003)